File: Modul 6/Soal 4/PRAK604-2410817210019-AchmadReihanAlfaiz.py
code = input()
message = input()

code = code.rstrip("\n")
message = message.rstrip("\n")

if len(code) != len(message):
    print("Panjang kalimat berbeda, pesan palsu")
    exit()

star_count = 0
hash_count = 0
result = []

for c, m in zip(code, message):
    if c.isspace() and m.isspace():
        result.append(" ")
    else:
        result.append("*" if c == m else "#")
        if c == m:
            star_count += 1
        else:
            hash_count += 1

print("".join(result))
print(f"* = {star_count}")
print(f"# = {hash_count}")
print("Pesan Asli" if star_count >= hash_count else "Pesan Palsu")

File: Modul 6/Soal 4/PRAK604-2410817210019-AchmadReihanAlfaiz.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main()
{
     char code[100], message[100];
     int starCount = 0, hashCount = 0;

     fgets(code, sizeof(code), stdin);
     fgets(message, sizeof(message), stdin);
     code[strcspn(code, "\n")] = 0;
     message[strcspn(message, "\n")] = 0;

     if (strlen(code) != strlen(message))
     {
          printf("Panjang kalimat berbeda, pesan palsu\n");
          return 0;
     }

     for (int i = 0; code[i]; i++)
     {
          if (isspace(code[i]) && isspace(message[i]))
               putchar(' ');
          else
               putchar(code[i] == message[i] ? '*' : '#'),
                   (code[i] == message[i]) ? starCount++ : hashCount++;
     }

     printf("\n* = %d\n# = %d\n%s\n", starCount, hashCount,
            starCount >= hashCount ? "Pesan Asli" : "Pesan Palsu");

     return 0;
}